### DISCUSSION TITLE
Can I compile it for TTGO Lora32 V1 B&W display ?

### DISCUSSION THREAD

**Discussion by BartDePauw** [BitMaker-hub/NerdMiner_v2#417] [General]
Don't have a E32 with color display but a (old) TTGO Lora32 V1 with B&W display
Think display is 128x64
Board can be found here : https://esp32-ttgo.blogspot.com/
I have the Arduino IDE in use, so can compile it myself.

Any help where to specify and how to set the B&W display ?
Can it uberhaupt work with a B&W display ?

Thanks for any reply and assistance !

**Comment by paelzer** [comment]
Hi
If the display on your mentioned board is an SSD1306 I2C type, then will work since I did a similar thing with a stand alone display on an ESP32 super mini couple of weeks ago. I set it up in VScode/Platformio.

You can use Adafruit's SSD1306 driver library to control the display part: https://github.com/adafruit/Adafruit_SSD1306.
But with 128x64 B/W of course one is limited when it comes to GUI. :-)

You will see how it's setup in this fork of the original NerdminerV2:
https://github.com/paelzer/NerdMiner_v2_paelzer

In Platformio you have to select "ESP32_super_mini_SSD1306" as env.

Boris

**Comment by BartDePauw** [comment]
Thanks for reply !

After flashing bootloader,partions,firmware with ESP32 Flash Download tool, ESP32 only boot with the 'ESP_SSD1306' compilation.
'ESP32_super_mini_SSD1306' gives boot loop.

But with 'ESP_SSD1306' compilation still no working display :
NerdMiner v2 starting......
SSD1306 ... display init
SSD1306 ... no loading screen

Do display pins need to be set somewhere before compiling ?

**Reply by paelzer** [reply]
Hello again

I just checked the specs of your device by the link you provided: https://esp32-ttgo.blogspot.com/

Accordingly the I2C pin defines for your display are:
SCL pin: 15
SDA pin: 4
RST pin: 16:

This pin defines are mentioned as "non-standard". 

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/83827311-a6cf-479d-9aa6-bd170b1bf458)

You can try the following:

In the file "src/drivers/displays/esp32_ssd1306_driver.cpp" add the following line to the "ssd1306_Display_Init" function:

Wire.setPins(4, 15);

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/d064d2ea-8139-44a0-998e-cd00ca6e17a0)

In the top of the file, for the OLED_RESET define replace value **-1** by **16**

![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/f66cf53a-c8be-40ed-ad20-c3c35d8c7b02)

Save the file.

To be on the save side, additionally in the file "src/drivers/devices/esp32_ssd1306.h" comment out the line:

#define PIN_BUTTON_1 0
![image](https://github.com/BitMaker-hub/NerdMiner_v2/assets/16374477/1cfd0b04-0269-4832-8285-8ae560c5f594)

Save the file.

Select the env "ESP32_SSD1306" in Platformio. Then compile and flash your board.

Would be interesting to know if this works.

Have fun!

Boris

**Comment by BartDePauw** [comment]
Thanks Boris !
& working 👍 
![Nerdminer v2 on TTGO Lora32 V1](https://github.com/BitMaker-hub/NerdMiner_v2/assets/31830190/7b890637-d3da-47c9-8fff-cc31aae7cf0a)

**Comment by paelzer** [comment]
Nice!! Glad I could help :-)
